src/main/java/hudson/plugins/active_directory/ActiveDirectoryUnixAuthenticationProvider.java

```diff
@@ -439,9 +440,12 @@ private String getPrincipalName(String username, String domainName) {
                     long start = System.nanoTime();
                     boolean found = false;
                     long duration = 0;
+
                     try {
                         found = chainGroupLookup(domainDN, userDN, context, groups);
                         duration = TimeUnit2.NANOSECONDS.toSeconds(System.nanoTime() - start);
+                    } catch (TimeLimitExceededException e) {
+                        LOGGER.log(Level.WARNING, "LDAP response read time out. AD will fall back to recursive lookup", e);
```

Comment:
This is not necessarily a specific "read timeout" - but could be a query exceeded an allowed time on the server side etc..